src/battle_script_commands.c

```diff
@@ -9851,29 +9883,56 @@ static void Cmd_handleballthrow(void)
                 break;
             case ITEM_HEAVY_BALL:
                 i = GetPokedexHeightWeight(SpeciesToNationalPokedexNum(gBattleMons[gBattlerTarget].species), 1);
-                if (i < 1024)
-                    ballMultiplier = 5;
-                else if (i < 2048)
-                    ballMultiplier = 10;
-                else if (i < 3072)
-                    ballMultiplier = 20;
-                else if (i < 4096)
-                    ballMultiplier = 30;
-                else
-                    ballMultiplier = 40;
+                #if P_HEAVY_BALL_MODIFIER >= GEN_7
+                    if (i < 1000)
```

Comment:
I think the Heavy Ball modifiers need to be reviewed. There are 3 sets. Gen 2's (which should be the default if Gen 3 is set), Gen 4's (if GEN_4-6 is set), and Gen 7+'s. They don't multiply or replace the 1 from what I'm looking at. They add or subtract instead. If the value is negative, it gets set to 1. 

Confirmation and implementation if confirmed would be appreciated.

Reply:
I agree. When I made the PR, I only took the values set by Egg initially, but it's better to rework it to match the main games.
And the config options for the different generations make sense, so I'll use those ranges.

Do I commit the implementation for review here, or do we see on Discord?